### PR TITLE
19 build header component emimint

### PR DESCRIPTION
**Pull request by Emimint**
# What changes did you make?

- Added the `.eslintcache` to `.gitignore`
- Added Figma image from Figma to `public` folder
- Created the Header component and its css
- Added mobile responsivity with Tailwind classes

# Why did you make these changes?

The Header is the first component that will be displayed and I assigned myself to do it.

# Screenshot of visual changes to the app (if applicable)

image

(PS: For now, the image covers the entire screen, but it is the only component so we can fix the css later)

I am waiting for your feedback 🙏 !

### CODE REVIEW

File: src/App.jsx
Comment (robester0403) [inline src/App.jsx]: Do we need a div here? Maybe use a fragment until you have more code later?

Comment (Emimint) [reply src/App.jsx]: @robester0403 actually I am not sure how we will go about that yet, as we will all need to have a discussion about that as the components are built, but I will keep that in mind.

File: src/components/header/Header.jsx
Comment (AOA19) [inline src/components/header/Header.jsx]: - The background image needs an overlay (similar to how it looks in the Figma file) so the words are easy to read. You could had to overlay to the <div> named "header"
- I couldn’t tell if the place-items-center class you put on this div is actually doing anything. When I removed it nothing changed. You already placed “text-center” on this div so I don’t think you need to any the place-items-center for the headers.

File: src/components/header/Header.jsx
Comment (AOA19) [inline src/components/header/Header.jsx:8]: The <h2> text gets very small between a min-width of 640px and a max-width of 767px. The font-size goes from 16px to 12px back to 16px as you size down. For accessibly reasons it should be 16px or higher so it's easy to read on smaller screens.

Comment (AOA19) [inline src/components/header/Header.jsx]: - The h2 text gets very small between a min-width of 640px and a max-width of 767px. The font size goes from 16px to 12px to 16px as you size down. For accessibility reasons it should be 16px or higher so it's easier to read on smaller screens.

File: src/components/header/Header.css
Comment (AOA19) [inline src/components/header/Header.css:1]: - @Emimint  The background image doesn’t scale very well when you size the page down. The image size it too large. Play around with the background-size and background-position properties 
- You have a bit of redundant code for the h1 and h2 , like the font-family, font-style, line-height, and text-shadow. You could create a class for both the h1 and h2 and put the similar styling under that class name so you only have to write it once.

**Comment by Emimint** [comment]
@AOA19 thanks a lot for the conselling 🙏 ! I have added the changes from @allison-truhlar as well (the search portion at the bottom). Please let me know if there is more changes to do.

image

**Comment by allison-truhlar** [comment]
@Emimint Thank you for working on the header component and for already addressing the feedback from Alexandra! 

I have four suggestions, only the first one of which I think is really necessary for merging this code, the others are just preference:

1. I would consider putting a height parameter on the header class. Maybe something like `max-height: 40vh` , eyeballing from the Figma design? My thinking is we don't want the header to fill the entire viewport.
2. If we limit the height of the header, then I would suggest adding `background-position: center` to ensure we capture the more interesting details of the background image.
3. It seems like a big jump to go from side padding of 0.5rem to 8rem at larger screen widths. I would usually handle this with a container div that I set at a percentage width of the body - usually around 90% for smaller screens and maybe closer to 70% for larger screens. The advantage to this is that it will scale across intermediate screen sizes. I would put this container around the h1 and h2, inside the header div. 
4. I would consider switching the order of the two media queries, so that as you read down the file, the queries are inclusive of the last one (so smaller screens => larger screens, since we're using min-width).

**Comment by allison-truhlar** [comment]
One additional change to request - the linter caught this, on line 9 of Header.css:
Files in the public directory are served at the root path.
Instead of /public/header_img.jpg, use /header_img.jpg.

**Comment by Emimint** [comment]
Hey @allison-truhlar . I updated the view-port height. As for the rest of the changes, how about we see once we put all the pieces of the app together? I think we can all discuss and agree once we get there :) !
image

**Review by allison-truhlar — APPROVED** [review]
@Emimint Thanks for adding the max-height for the header - we can always adjust later if we want something different. Thanks also for removing the .jsx on the App.jsx import in main.jsx, flagged by the linter.

I agree - the rest of my comments can wait until later! I approve this code as-is for merging.

**Pull request merged**
